### PR TITLE
Add multi mode

### PR DESCRIPTION
**Pull request by sfiligoi**

### CODE REVIEW

File: src/api.hpp [hunk 1]
```diff
@@ -360,6 +360,43 @@ EXTERN ComputeStatus unifrac_to_file(const char* biom_filename, const char* tree
                                      bool bypass_tips, unsigned int n_substeps, const char* format,
                                      unsigned int pcoa_dims, const char *mmap_dir);
 
+/* Compute UniFrac with subsampling several times and save to file
+ *
+ * biom_filename <const char*> the filename to the biom table.
+ * tree_filename <const char*> the filename to the correspodning tree.
```
Comment (wasade) [inline src/api.hpp:366]:
```suggestion
 * tree_filename <const char*> the filename to the corresponding tree.
```

File: src/testdata/compare_unifrac_stats_multi.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  if (d['stat_n_groups'][i]!=n_groups):
+    print("n_groups does not match for %i:"%i, d['stat_n_groups'][i],n_groups)
+    sys.exit(1)
+  if (d['stat_n_permutations'][i]!=n_perms):
+    print("n_perms does not match for %i:"%i,d['stat_n_permutations'][i],n_perms)
+    sys.exit(1)
+
+  t1=d['stat_values'][i]
+  t2=value
+  m=abs((t1-t2)/t2)
```
Comment (wasade) [inline src/testdata/compare_unifrac_stats_multi.py:40]: Why div by t2?

**Review by wasade — COMMENTED** [review]
Very minor things

File: src/testdata/compare_unifrac_stats_multi.py (hunk 2 above)
Comment (sfiligoi) [reply src/testdata/compare_unifrac_stats_multi.py:40]: Relative difference.

File: src/api.cpp [hunk 3]
```diff
@@ -1109,6 +1049,408 @@ inline herr_t write_hdf5_array2D(hid_t output_file_id, hid_t real_id,
...
+         free(proportion_explained);
+         free(samples);
+         if (rc!=write_okay) throw "PCOA write failed";
+         TDBG_STEP("pcoa saved")
+      } // if pcoa
+
+      n_results++;
+   }
+
+   void write_stats(unsigned int           stat_n_vals,
```
[the remark this answers is not in the document]
Comment (sfiligoi) [reply src/api.cpp:1177]: It's the same spacing I was using in write_mat_from_matrix_hdf5_T.

**Pull request merged**